Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,782 @@
+            $currentForegroundColor = $host.ui.RawUI.ForegroundColor
+            if ( $_.Exception.Message -eq "The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.") {
+                $host.ui.RawUI.ForegroundColor = "Red"
+                Write-Output $Message
                 $host.ui.RawUI.ForegroundColor = "Yellow"
-                Write-Output "If you are using Microsoft Defender then AMSI may be disabled or you are using a AntiVirus Product that may not be AMSI capable (Please Check with your AntiVirus Provider for Exchange AMSI Support)"
+                Write-Output "You could use the -IgnoreSSL parameter"
                 $host.ui.RawUI.ForegroundColor = $currentForegroundColor
-                Write-Output $msgNewLine
-            } else {
-                Write-Output $msgNewLine
+            } elseif ($_.Exception.Message -eq "The remote server returned an error: (400) Bad Request.") {
                 $host.ui.RawUI.ForegroundColor = "Green"
                 Write-Output "We sent an test request to the ECP Virtual Directory of the server requested"
-                $host.ui.RawUI.ForegroundColor = "Red"
+                $host.ui.RawUI.ForegroundColor = "Yellow"
                 Write-Output "The remote server returned an error: (400) Bad Request"
                 $host.ui.RawUI.ForegroundColor = "Green"
-                Write-Output "---------------------------------------------------------------------------------------------------------------"
+                $bar = ""
+                1..($Host.UI.RawUI.WindowSize.Width) | ForEach-Object { $bar += "-" }
+                Write-Output $bar
                 $host.ui.RawUI.ForegroundColor = "Yellow"
                 Write-Output "This may be indicative of a potential block from AMSI"
                 $host.ui.RawUI.ForegroundColor = "Green"
-                $msgCheckLogs = "Check your log files located in " + $installPath + "Logging\HttpRequestFiltering\"
+                $msgCheckLogs = "You can check your log files located in %ExchangeInstallPath%\Logging\HttpRequestFiltering\"
                 Write-Output $msgCheckLogs
-                $msgDetectedTimeStamp = "for a Detected result around " + $dateTime.ToUniversalTime()
+                $host.ui.RawUI.ForegroundColor = $currentForegroundColor
+                $msgDetectedTimeStamp = "You should find result around $((Get-Date).ToUniversalTime().ToString("M/d/yyy h:mm:ss tt")) UTC"
                 Write-Output $msgDetectedTimeStamp
+                if ( $isServer ) {
+                    Write-Output ""
+                    Write-Host "Checking logs on $server at $($testTime.ToString("M/d/yyy h:mm:ss tt"))"
+                    $HttpRequestFilteringLogFolder = $null
+                    if ( $server.ToLower() -eq $env:COMPUTERNAME.ToLower() ) {
+                        $HttpRequestFilteringLogFolder = Join-Path $env:ExchangeInstallPath "Logging\HttpRequestFiltering\"
```

Comment:
It will execute that line only if you are running the script in the same server, so it should local.
is that what you mean?
It is working in my tests

Reply:
It won't work if you launch PowerShell.exe and run the script. `Confirm-ExchangeShell` will load EMS, and it won't set that variable as far as I know.

Reply:
In all test that I did it works, including in a powershell before and after load `Confirm-ExchangeShell`:
![image](https://github.com/microsoft/CSS-Exchange/assets/103440830/7b260837-2f7b-4732-addf-0dab85f9310e)
As it is a Environment variable and that line is only executed in case that we are in the expected Exchange server (locally) I think it should be available in any case. I am assuming that all Exchange server has the ExchangeInstallPath variable. am I wrong?
In any case, if you think it is better, I can use `HKLM:\SOFTWARE\Microsoft\ExchangeServer\v15\Setup\MsiInstallPath` instead.

Reply:
Wasn't aware of this, not sure if this didn't work in order OS/PowerShell. But it definitely works now so we can keep it as long as we are on the Exchange Server.